packages/matter.js/src/behavior/system/network/ServerNetworkRuntime.ts

```diff
@@ -81,22 +81,56 @@ export class ServerNetworkRuntime extends NetworkRuntime {
     }
 
     /**
-     * The IPv6 {@link UdpInterface}.  We create this interface independently of the server so the OS can select a port
-     * before we are fully online.
+     * The {@link UdpInterface} instances we listen on.  We create these independently of the server so the OS can
+     * select a port before we are fully online.
      */
-    protected async getPrimaryNetInterface() {
-        if (this.#primaryNetInterface === undefined) {
-            const port = this.owner.state.network.port;
-            this.#primaryNetInterface = await UdpInterface.create(
-                this.owner.env.get(Network),
-                "udp6",
-                port ? port : undefined,
-                this.owner.state.network.listeningAddressIpv6,
-            );
-
-            await this.owner.set({ network: { operationalPort: this.#primaryNetInterface.port } });
+    protected async getTransports() {
+        if (this.#transports !== undefined) {
+            return this.#transports;
         }
-        return this.#primaryNetInterface;
+
+        const transports = new Set<TransportInterface>();
+        const bleTransports = new Set<TransportInterface>();
+
+        for (const address of this.owner.state.network.listen) {
+            switch (address.protocol) {
+                case "udp":
+                case "udp4":
+                case "udp6":
+                    const udp = await UdpInterface.create(
+                        this.owner.env.get(Network),
+                        address.protocol,
+                        address.port,
+                        address.address,
+                    );
+
+                    transports.add(udp);
+
+                    // We advertise the first (most likely only) UDP port we open as our official address
+                    await this.owner.act(agent => {
+                        const state = agent.network.state;
+                        if (state.operationalPort === -1) {
+                            state.operationalPort = udp.port;
+                        }
+                    });
+
+                    break;
+
+                case "ble":
+                    // TODO - HCI ID and/or multiple BLE transports?  (Former seems valuable, latter not really but
```

Comment:
just a single BLE is making sense. forbid multiple

Reply:
... and see above.formally only one BLE per host ... soooo multiple servers the server should define if ble should be (not) added too?